020 final qa (#134)
spatialos/online-services: docs/content/services-packages/analytics-pipeline/usage.md

| `event_time` | Optional | Defaults to the current UTC time period. If needed, you can set a time period that overrides this, must be one of `0-8`, `8-16` or `16-24`. |
| `session_id` | Optional | Should be set, otherwise defaults to `session-id-not-available`. |

The parameters in the table above (except for `key`) play a part in how the data ends up in the GCS bucket (denoted `{{in_double_curly_brackets}}`):

> gs://[[gcs_bucket_name]]/data_type=[[data_type]]/analytics_environment={{analytics_environment}}/event_category={{event_category}}/event_ds={{event_ds}}/event_time={{event_time}}/{{session_id}}/[[timestamp]]-[[random_alphanum]]

> Note that the endpoint will automatically (denoted `[[in_double_square_brackets]]`):

Reviewer: I'm not sure what 

> (denoted `[[in_double_square_brackets]]`)

 means here.

Author: The things the endpoint takes care of automatically are in double square brackets. See the example path right above it, there's two types of parameter, `[[]]` (always automatic) and `{{}}` (configurable).

Author: I'll make this clearer

| `event_time` | Optional | Defaults to the current UTC time period. If needed, you can set a time period that overrides this, must be one of `0-8`, `8-16` or `16-24`. |
| `session_id` | Optional | Should be set, otherwise defaults to `session-id-not-available`. |

> The parameters in the table above (except for `key`) play a part in how the data ends up in the GCS bucket (denoted `{{in_double_curly_brackets}}`):

Reviewer: ```suggestion
The parameters in the table above (except for `key`) play a part in how the data ends up in the GCS bucket. In the example below, these parameters are denoted by `{{double_curly_brackets}}`):
```

> gs://[[gcs_bucket_name]]/data_type=[[data_type]]/analytics_environment={{analytics_environment}}/event_category={{event_category}}/event_ds={{event_ds}}/event_time={{event_time}}/{{session_id}}/[[timestamp]]-[[random_alphanum]]

Note that the endpoint will automatically:

* use the `[[gcs_bucket_name]]` of the analytics cloud storage bucket we created with Terraform.
* determine `[[data_type]]`, which can either be `json` (when valid JSON is `POST`ed) or `unknown` (otherwise).
* set the fields `[[timestamp]]` and `[[random_alphanum]]` (a random string to avoid collisions) as well.

> Parameters in the example storage path above that are always taken care of automatically by the endpoint are denoted `[[in_double_square_brackets]]`.

Reviewer: Thanks! 

I initially thought the example above made sense, but now I see that it means "the parameters are denoted in double curly brackets" (whereas it sounds like it's saying "the GCS bucket is denoted in double curly brackets"). I've suggested an update (above) to make it clearer.

And for this one, I'll just suggest a minor change in phrasing:

```suggestion
Parameters in the example storage path above that the endpoint deals with automatically are denoted by `[[double_square_brackets]]`.
```

Starting the `--data` value with the `@` symbol means you are passing it a file.

> <%(#Expandable title="Want to instead pass the <code>--data</code> payload as a string?")%>

Reviewer: Good spot!

Reviewer: I should have said this earlier, but it would sound better as:

> Want to pass the <code>--data</code> payload as a string instead?

"instead" --> end

Author: Adopted

To test this yourself, replace:

* `{{local_path_json_payload}}` with the local path of the JSON file you just created.
* `{{your_google_project_id}}` and `{{your_analytics_api_key}}` (created in [step 3.1 of the deploy section]({{urlRoot}}/content/services-packages/analytics-pipeline/deploy#3-1-store-your-secret)) with your own values.
* `{{analytics_environment}}` with `testing`.
* `{{session_id}}` with any made up session identifier (such as `f58179a375290599dde17f7c6d546d78`).

Then, navigate to [BigQuery](https://console.cloud.google.com/bigquery) and submit the following queries to check your results:

> 

Reviewer 2: Don't you want the sql syntax highlighting?

Author: I very much do, but it does not work in our docs.. Leaving it out currently looks better.
